src/ui/main.qml

```diff
@@ -348,6 +348,82 @@ Window {
         }
     }
 
+    Connections {
+        target: VPNErrorHandler
+        function onSubscriptionGeneric() {
+            mainStackView.push("qrc:/ui/views/ViewErrorFullScreen.qml", {
+                // Problem confirming subscription...
+                headlineText: VPNl18n.GenericPurchaseErrorGenericPurchaseErrorHeader,
+
+                // An unexpected error has occurred, please try again.
+                // Visit our help center below to learn how to troubleshoot this issue or contact support.
+                errorMessage: VPNl18n.GenericPurchaseErrorGenericPurchaseErrorText,
+
+                // Try again
+                buttonText: VPNl18n.GenericPurchaseErrorGenericPurchaseErrorButton,
+                buttonObjectName: "errorTryAgainButton",
+                buttonOnClick: mainStackView.pop,
+                signOffLinkVisible: false,
+                getHelpLinkVisible: true
+            });
+        }
+
+        function onNoSubscriptionFound() {
+            mainStackView.push("qrc:/ui/views/ViewErrorFullScreen.qml", {
+                // Problem confirming subscription...
+                headlineText: VPNl18n.GenericPurchaseErrorGenericPurchaseErrorHeader,
+
+                // Sorry we are unable to find your subscription, please try again.
+                // Visit our help center below to learn more about how to troubleshoot our VPN.
+                errorMessage: VPNl18n.RestorePurchaseNotFoundErrorRestorePurchaseNotFoundErrorText,
+
+                // Try again
+                buttonText: VPNl18n.GenericPurchaseErrorGenericPurchaseErrorButton,
+                buttonObjectName: "errorTryAgainButton",
+                buttonOnClick: mainStackView.pop,
+                signOffLinkVisible: true,
+                getHelpLinkVisible: true
+            });
+        }
+
+        function onSubscriptionExpired() {
+            mainStackView.push("qrc:/ui/views/ViewErrorFullScreen.qml", {
+                // Problem confirming subscription...
+                headlineText: VPNl18n.GenericPurchaseErrorGenericPurchaseErrorHeader,
+
+                // Sorry we are unable to connect your Firefox Account to a current subscription.
+                // Please try again or contact our support team for further assistance.
+                errorMessage: VPNl18n.RestorePurchaseExpiredErrorRestorePurchaseExpiredErrorText,
+
+                // Try again
+                buttonText: VPNl18n.GenericPurchaseErrorGenericPurchaseErrorButton,
+                buttonObjectName: "errorTryAgainButton",
+                buttonOnClick: mainStackView.pop,
+                signOffLinkVisible: false,
```

Comment:
Are they signed in at this point, but without a subscription? If so, we probably need to make `signOffLinkVisible` true so that they can maybe sign out and try signing in with a different Firefox Account.

Reply:
We can make it visible but it is not part of the spec.